In [1]:
import numpy as np

# Given covariance matrix
cov_matrix = np.array([[5, 2],
                        [2, 2]])

# Step 1: Calculate the eigenvalues and eigenvectors of the covariance matrix
eigenvalues, eigenvectors = np.linalg.eig(cov_matrix)

# Step 2: Sort the eigenvalues and eigenvectors in descending order
sorted_indices = np.argsort(eigenvalues)[::-1]
eigenvalues = eigenvalues[sorted_indices]
eigenvectors = eigenvectors[:, sorted_indices]

# Step 3: Calculate the proportion of total variance explained by each principal component
total_variance = np.trace(cov_matrix)
variance_explained = eigenvalues / total_variance

# Step 4: Calculate the principal components (Y1 and Y2)
# Y1 is the first principal component (eigenvector corresponding to the largest eigenvalue)
Y1 = eigenvectors[:, 0]
# Y2 is the second principal component (eigenvector corresponding to the second largest eigenvalue)
Y2 = eigenvectors[:, 1]

# Print the results
print("Principal Component Y1:", Y1)
print("Principal Component Y2:", Y2)
print("Proportion of Variance Explained by Y1:", variance_explained[0])


Principal Component Y1: [0.89442719 0.4472136 ]
Principal Component Y2: [-0.4472136   0.89442719]
Proportion of Variance Explained by Y1: 0.8571428571428571


In [2]:
eigenvalues

array([6., 1.])

Proportion of variance explained by principal component i = eigenvalue_i / sum(eigenvalues)

- Proportion of variance explained by Y1 = 7 / (7 + 3) = 85.71%
- Proportion of variance explained by Y2 = 3 / (7 + 3) = 14.29%

Therefore, the first principal component, Y1, explains 85.71% of the variance in the data, while the second principal component, Y2, explains 14.29% of the variance in the data. This suggests that a significant portion of the dataset's variability can be captured by considering only the first principal component.

Also, the first principal component has a much larger eigenvalue than the second principal component. This means that the first principal component captures much more of the variance in the data than the second principal component.

__Interesting Findings__: 
The fact that Y1 explains a significant portion of the variance implies that the original variables in the dataset are correlated and can be efficiently represented using just one component. This could indicate a high degree of redundancy or multicollinearity among the variables.